pgjdbc/src/main/java/org/postgresql/ssl/jdbc4/LibPQFactory.java

```diff
@@ -251,6 +255,22 @@ public boolean verify(String hostname, SSLSession session) {
     }
     // Extract the common name
     X509Certificate serverCert = peerCerts[0];
+
+    try {
+      // Check for Subject Alternative Names (see RFC 6125)
+      Collection subjectAltNames = serverCert.getSubjectAlternativeNames();
+      Iterator sanIt = subjectAltNames.iterator();
+      while (sanIt.hasNext()) {
+        List list = (List) sanIt.next();
+        String san = ((String) list.get(1));
+        if (san.equals(hostname)) {
```

Comment:
OkHttp has a slightly different code: https://github.com/square/okhttp/blob/master/okhttp/src/main/java/okhttp3/internal/tls/OkHostnameVerifier.java#L97-L106

I think we might want to check "type" field as well.